File: 0701_0800/0733/0733.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    if n == 1:
        q.write(f"1\n{f.readline()}")
    else:
        p = [int(x) for x in f.readline().split()]
        a = [None for i in range(n)]
        a[n - 1] = p[n - 1]
        for i in range(n - 2, 0, -1):
            a[i] = p[i] if p[i] > a[i + 1] else a[i + 1]
        s = [p[0]]
        m = p[0]
        for i in range(1, n - 1):
            if p[i] > m:
                m = p[i]
            if p[i] >= m or p[i] >= a[i]:
                s.append(p[i])
        s.append(p[n - 1])
        q.write(str(len(s)) + '\n' + ' '.join(str(x) for x in s))

File: 0701_0800/0733/0733.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, *p, *a, d = 0;

	fscanf(f, "%d", &n);
	p = (int*)malloc(sizeof(int) * n);
	a = (int*)malloc(sizeof(int) * n);
	for (int i = 0; i < n; ++i)
		fscanf(f, "%d", p + i);
	a[n - 1] = p[n - 1];
	for (int i = n - 2; i > 0; --i)
		a[i] = (p[i] > a[i + 1] ? p[i] : a[i + 1]);
	a[d++] = p[0];
	for (int i = 1, m = p[0]; i < n - 1; ++i)
	{
		if (p[i] > m)
			m = p[i];
		if (p[i] >= m || p[i] >= a[i])
			a[d++] = p[i];
	}
	if (n > 1)
		a[d++] = p[n - 1];
	fprintf(q, "%d\n", d);
	for (int i = 0; i < d; ++i)
		fprintf(q, "%d ", a[i]);
	return 0;
}
